Mathlib/Probability/Distributions/Exponential.lean

```diff
@@ -0,0 +1,251 @@
+    fun x ↦ ENNReal.toReal (∫⁻ x in Iic x, exponentialPdf r x) := by
+  ext x
+  simp only [exponentialPdf, exponentialCdfReal, cdf_eq_toReal]
+  simp only [expMeasure, measurableSet_Iic, withDensity_apply]
+  rfl
+
+open Topology
+
+lemma antiDeriv_expDeriv_pos {r x : ℝ} :
+    HasDerivAt (fun a ↦ -1 * exp (-(r * a))) (r * exp (-(r * x))) x := by
+  convert (((hasDerivAt_id x).const_mul (-r)).exp.const_mul (-1)) using 1
+  · simp only [id_eq, neg_mul]
+  simp only [id_eq, neg_mul, mul_one, mul_neg, one_mul, neg_neg, mul_comm]
+
+lemma lint_eq_antiDeriv (r : ℝ) (hr : 0 < r) : ∀ x : ℝ,
+    (∫⁻ y in Iic x, exponentialPdf r y =
+    ENNReal.ofReal (if 0 ≤ x then 1 - exp (-(r * x)) else 0)) := by
+  intro x'
+  split_ifs with h
+  case neg =>
+    simp only [exponentialPdf_eq]
+    rw [set_lintegral_congr_fun measurableSet_Iic, lintegral_zero, ENNReal.ofReal_zero]
+    exact ae_of_all _ fun a (_ : a ≤ _) ↦ by rw [if_neg (by linarith), ENNReal.ofReal_eq_zero]
+  case pos =>
+    rw [lintegral_split_bounded _ h, lintegral_nonpos (le_refl 0), zero_add]
+    simp only [exponentialPdf_eq]
+    rw[set_lintegral_congr_fun measurableSet_Icc (ae_of_all _
+        (by intro a ⟨(hle : _ ≤ a), _⟩; rw [if_pos hle]))]
+    rw [←ENNReal.toReal_eq_toReal _ ENNReal.ofReal_ne_top, ←integral_eq_lintegral_of_nonneg_ae
+        (eventually_of_forall fun _ ↦ le_of_lt (mul_pos hr (exp_pos _)))]
+    have : ∫ a in uIoc 0 x', r * rexp (-(r * a)) = ∫ a in (0)..x', r * rexp (-(r * a)) := by
+      rw [intervalIntegral.intervalIntegral_eq_integral_uIoc, smul_eq_mul, if_pos h, one_mul]
+    rw [integral_Icc_eq_integral_Ioc, ← uIoc_of_le h, this]
+    rw [intervalIntegral.integral_eq_sub_of_hasDeriv_right_of_le h
+      (f := fun a ↦ -1 * rexp (-(r * a))) _ _]
+    rw [ENNReal.toReal_ofReal_eq_iff.2 (by norm_num; positivity)]
+    · norm_num; ring
+    · simp only [intervalIntegrable_iff, uIoc_of_le h]
+      apply Integrable.const_mul (exp_neg_integrableOn_Ioc hr)
+    · have : Continuous (fun a ↦ rexp (-(r * a))) := by
+        simp only [← neg_mul]; exact  (continuous_mul_left (-r)).exp
+      exact Continuous.continuousOn (Continuous.comp' (continuous_mul_left (-1)) this)
+    · exact fun _ _ ↦ HasDerivAt.hasDerivWithinAt antiDeriv_expDeriv_pos
+    · apply Integrable.aestronglyMeasurable (Integrable.const_mul _ _)
+      rw [← integrableOn_def, integrableOn_Icc_iff_integrableOn_Ioc]
+      exact exp_neg_integrableOn_Ioc hr
+    · refine ne_of_lt (IntegrableOn.set_lintegral_lt_top ?_)
+      rw [integrableOn_Icc_iff_integrableOn_Ioc]
+      apply Integrable.const_mul (exp_neg_integrableOn_Ioc hr)
+
+/-- The Definition of the CDF equals the known Formular ``1 - exp (-(r * x))``-/
+lemma exponentialCdf_eq {r : ℝ} [Fact (0 < r)] : exponentialCdfReal r =
+    fun x ↦ if 0 ≤ x then 1 - exp (-(r * x)) else 0 := by
+  rw[expCdf_eq_lintegral]; ext x; rw [lint_eq_antiDeriv _ Fact.out]
+  rw[ENNReal.toReal_ofReal_eq_iff]
+  split_ifs with h <;> simp [mul_nonneg (le_of_lt Fact.out) _]
```

Comment:
This `simp` does not close all goals, which make it a non-terminal simp. Those are bad, because what simp does depends on which lemmas are tagged simp in the whole library. It might be that in the future, that simp suddenly simplifies to something slightly different and then the next line won't work, and it will be difficult to reconstruct what it should have simplified to.
That's why we should replace all non-terminal simp by `simp only` (use `simp?` and replace with the suggestion).